src/value.cpp

```diff
@@ -564,6 +564,23 @@ void MemRef::setWritable(bool writable) {
   m->setWritable(bid, writable);
 }
 
+MemRef::Layout MemRef::toSubViewLayout(
+   const vector<z3::expr> &offsets,
+   const vector<z3::expr> &strides) {
+   // Before : <(d0, d1) -> (d0 * s0 + d1)>,
+   // After: <(d0, d1) -> ((d0 + offsets[0]) * strides[0] * s0 + (d1 + offsets[1]) * strides[1])>
```

Comment:
I have two questions:
1. Do you have any idea why it is not `((d0 + offsets[0] * strides[0]) * s0 + (d1 + offsets[1] * strides[1]))`?
2. The document says: `strides: memref-rank number of strides that compose multiplicatively with the base memref strides in each dimension.` Is this consistent with your implementation?

Reply:
As mentioned in second question, strides should "compose multiplicatively with the base memref strides".
In above example, d0's original stride is "s0" and d1's original stride is "1".
So after subview operation, these strides should be multiplied with strides[0], strides[1] values, resulting in "s0 * strides[0]" as d0's and "1 * strides[1]" as d1's strides